core/src/main/java/com/linecorp/armeria/internal/InternalRequestContext.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal;
+
+import com.linecorp.armeria.common.RequestContext;
+
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.UnpooledByteBufAllocator;
+
+/**
+ * An interface defining internal-only methods of a {@link com.linecorp.armeria.common.RequestContext}. These
+ * methods should be used with care, generally only in generic services and protocol implementations, not user
+ * business logic.
+ */
+public interface InternalRequestContext {
```

Comment:
The reason the build failed is that some of the public classes now implement an interface in the internal package, exposing the internal types in the type hierarchy. I think it's not desirable to expose an internal class in the type hierarchy, but I do not have a clear fix for this yet. Let me think a little bit about it.